src/docs/background/10-custom-pricing/index.md

```diff
@@ -0,0 +1,263 @@
+slug: custom-pricing
+updated: 2019-03-18
+category: background
+ingress:
+  This article introduces you to the concept of custom pricing and how
+  it differs from other pricing methods.
+published: true
+---
+
+## What kind of pricing can you achieve with Flex
+
+Service marketplaces usually base their pricing on the length of a
+booking, on a number of booked units, or the combination of these two.
+With Flex you can design your pricing using these two parameters but the
+pricing can also be extended to support more complicated business models
+or even be replaced with a completely different model altogether. Here
+are some examples of common pricing schemes you might want to add on top
+of the regular listing price.
+
+- **Add-ons.** Upsell additions on top of the regular price. Examples:
+  cleaning fee, insurance, delivery.
+- **Discount based on booking length.** An example: daily price is
+  $20, weekly
+  price $70, and monthly price \$200. Another example: 20%
+  discount on daily rate for bookings of 5 or more days, 30% discount on
+  bookings of 10 or more days, and so on.
+- **Taxes.** Add any type of tax rates to the listing price and display
+  them as separate line items in the receipt.
+- **Offer multiple pricing packages in the same listing.** An example: a
+  hairdresser offers both haircuts and hair dying.
+- **Seasonal pricing.** Examples: weekends cost more than weekdays,
+  summers cost more than winters, and so on.
+- **Quantity discount.** An example: booking a room for two people costs
+  $100,
+  and after that each additional person costs $20 extra.
+
+All of these can be achieved using _custom pricing_. Custom pricing
+provides the marketplace website or app with a wide range of ways to
+affect how bookings are created. In this article we'll look at bit
+closer on how pricing works in Flex and how it can be extended with
+custom pricing.
+
+## How pricing works in Flex
+
+In Flex, the total price of a transaction is defined by its _line
+items_. Line items describe what is included in a transaction. It
+depends on the transaction process what the line items contain but it
+can be a varying set of things from number of booked units to customer
+and provider commissions and payment refunds. The line items of a
+transaction depend on the _actions_ that are used in the transaction
+process in the transition(s) in which the transaction is created. In
+general, actions can perform anything that is related to the transaction
+process like make payments and send notifications. But when it comes to
+pricing, it's up to the actions to construct the right set of line items
+based on the request parameters that are passed to the API.
+
+Different actions can be used to create different pricing schemes. A
+night-based pricing action calculates the number of nights between the
+start and end of the booking to calculate the number of booked units
+whereas a day-based action respectively calculates the number of days. A
```

Comment:
Not the most perfect example since days and nights calculation is exactly the same and the only difference is the line item code we use. :)